### DISCUSSION TITLE
reading JSON from a file.

### DISCUSSION THREAD

**Discussion by Blackqx0** [nlohmann/json#3708] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/102927513/185728071-4b7bd2b7-7280-461e-83fa-b5feccaaedf2.png)
i get this error when i try to read json from a file, and it crashes me.

code:

> #include <fstream>
#include <nlohmann/json.hpp>
#include <iostream>
using json = nlohmann::json;
int main()
{
	std::ifstream file("Test.json");
	json j = json::parse(file);
	std::cout << j["message"] << std::endl;
	return 0;
}

**Comment by falbrechtskirchinger** [comment, marked as answer]
* Was the file opened correctly? Check `file.good()`.
* What's the exact error? Catch the exception and print its message.
* Does the file contain valid JSON and only a single value? (One object, array, string, ...)

**Reply by Blackqx0** [reply]
Thank You. my bad <3.
